src/unit_tests/syscheckd/test_fim_diff_changes.c

```diff
@@ -0,0 +1,2020 @@
+    configuration->diff_size_limit = 1024;
+    syscheck.comp_estimation_perc = 0.4;
+
+    expect_fim_diff_registry_tmp("queue/diff/tmp", "queue/diff/tmp/[x64] b9b175e8810d3475f15976dd3b5f9210f3af66043f17670fd80d6563a3d4283adfe14140907b75b0", (FILE *)1234, value_data);
+
+    expect_fim_diff_check_limits("queue/diff/tmp/[x64] b9b175e8810d3475f15976dd3b5f9210f3af66043f17670fd80d6563a3d4283adfe14140907b75b0", COMPRESS_FOLDER_REG, 2);
+
+    expect_string(__wrap__mdebug2, formatted_msg, "(6350): The estimation of the file size 'HKEY_LOCAL_MACHINE\\Software\\Classes\\batfile\\valuename' exceeds the disk_quota. Operation discarded.");
+
+    expect_string(__wrap_rmdir_ex, name, "queue/diff/tmp");
+    will_return(__wrap_rmdir_ex, 0);
+
+    char *diff_str = fim_registry_value_diff(key_name, value_name, value_data, data_type, configuration);
+
+    assert_ptr_equal(diff_str, NULL);
+}
+
+void test_fim_registry_value_diff_uncompress_fail(void **state) {
+    const char *key_name = "HKEY_LOCAL_MACHINE\\Software\\Classes\\batfile";
+    const char *value_name = "valuename";
+    const char *value_data = "value_data";
+    DWORD data_type = REG_EXPAND_SZ;
+    registry *configuration = &syscheck.registry[0];
+
+    expect_fim_diff_registry_tmp("queue/diff/tmp", "queue/diff/tmp/[x64] b9b175e8810d3475f15976dd3b5f9210f3af66043f17670fd80d6563a3d4283adfe14140907b75b0", (FILE *)1234, value_data);
+
+    expect_fim_diff_check_limits("queue/diff/tmp/[x64] b9b175e8810d3475f15976dd3b5f9210f3af66043f17670fd80d6563a3d4283adfe14140907b75b0", COMPRESS_FOLDER_REG, 0);
+
+    expect_w_uncompress_gzfile("queue/diff/registry/[x64] b9b175e8810d3475f15976dd3b5f9210f3af6604/3f17670fd80d6563a3d4283adfe14140907b75b0/last-entry.gz", "queue/diff/tmp/tmp-entry", (FILE *)1234);
+
+    expect_fim_diff_create_compress_file("queue/diff/tmp/[x64] b9b175e8810d3475f15976dd3b5f9210f3af66043f17670fd80d6563a3d4283adfe14140907b75b0", "queue/diff/tmp/tmp-entry.gz", 0);
+
+    expect_mkdir_ex(COMPRESS_FOLDER_REG, 0);
+
+    expect_save_compress_file("queue/diff/tmp/tmp-entry.gz", "queue/diff/registry/[x64] b9b175e8810d3475f15976dd3b5f9210f3af6604/3f17670fd80d6563a3d4283adfe14140907b75b0/last-entry.gz", 0);
+
+    expect_string(__wrap_rmdir_ex, name, "queue/diff/tmp");
+    will_return(__wrap_rmdir_ex, 0);
+
+    char *diff_str = fim_registry_value_diff(key_name, value_name, value_data, data_type, configuration);
+
+    assert_ptr_equal(diff_str, NULL);
+}
+
+void test_fim_registry_value_diff_create_compress_fail(void **state) {
+    const char *key_name = "HKEY_LOCAL_MACHINE\\Software\\Classes\\batfile";
+    const char *value_name = "valuename";
+    const char *value_data = "value_data";
+    DWORD data_type = REG_EXPAND_SZ;
+    registry *configuration = &syscheck.registry[0];
+
+    expect_fim_diff_registry_tmp("queue/diff/tmp", "queue/diff/tmp/[x64] b9b175e8810d3475f15976dd3b5f9210f3af66043f17670fd80d6563a3d4283adfe14140907b75b0", (FILE *)1234, value_data);
```

Comment:
We can replace these long static strings by using a couple of define:
```c
#define KEY_NAME_HASHED "b9b175e8810d3475f15976dd3b5f9210f3af6604"
#define VALUE_NAME_HASHED "3f17670fd80d6563a3d4283adfe14140907b75b0"
```

We can then do things like:
```suggestion
    expect_fim_diff_registry_tmp("queue/diff/tmp", "queue/diff/tmp/[x64] " KEY_NAME_HASHED VALUE_NAME_HASHED, (FILE *)1234, value_data);
```